### PR TITLE
Prevent vault of different collateral to be added

### PR DESCRIPTION
**Pull request by iamsahu**
Fix for #351 

## Details
A check was added in the `addVault` function to prevent a vault for a different ilk to be added

## Tests
A test was added in the `092_convex_wrapper` to check if the vault addition fails if a different vault is added.

### CODE REVIEW

**Comment by alcueca** [comment]
Something seems off with the single quotes and the TODOs, would you check you branched off the latest code?

File: test/092_convex_wrapper.ts [hunk 1]
```diff
@@ -304,4 +304,19 @@ describe('Convex Wrapper', async function () {
...
+    var join = await ladle.joins(CVX3CRV)
+    await convexWrapper.point(join)
+    // Batch action to build a vault & add it to the wrapper
+    const addVaultCall = convexLadleModule.interface.encodeFunctionData('addVault', [
+      convexWrapper.address,
+      '0x000000000000000000000000',
+    ])
+    await expect(
+      ladle.batch([ladle.buildAction(seriesId, USDC), ladle.moduleCallAction(convexLadleModule.address, addVaultCall)])
+    ).to.be.revertedWith('revert Vault is for different ilk')
```
Comment (alcueca) [inline test/092_convex_wrapper.ts:320]: The error message is different:
"Vault is for different ilk"
'revert Vault is for different ilk'

Is this test ok?

**Review by alcueca — COMMENTED** [review]
Looks good, but I'm concerned on the test giving a false positive. Would you please check?

File: test/092_convex_wrapper.ts (hunk 1 above)
Comment (iamsahu) [reply test/092_convex_wrapper.ts:320]: Really strange that it is passing. Let me investigate.

Comment (iamsahu) [reply test/092_convex_wrapper.ts]: The above test is failing locally. After changing the message to reflect what is in the contract is making it pass.

**Review by alcueca — APPROVED** [review]
It's concerning that our CI doesn't catch all test errors, though :/

Maybe moving to foundry would fix this.

**Pull request merged**